tsdb/ooo_head_read.go

```diff
@@ -341,14 +349,15 @@ func NewOOOCompactionHead(ctx context.Context, head *Head) (*OOOCompactionHead,
 			continue
 		}
 
-		mmapRef := ms.mmapCurrentOOOHeadChunk(head.chunkDiskMapper)
-		if mmapRef == 0 && len(ms.ooo.oooMmappedChunks) > 0 {
+		mmapRefs := ms.mmapCurrentOOOHeadChunk(head.chunkDiskMapper)
+		if len(mmapRefs) == 0 && len(ms.ooo.oooMmappedChunks) > 0 {
 			// Nothing was m-mapped. So take the mmapRef from the existing slice if it exists.
-			mmapRef = ms.ooo.oooMmappedChunks[len(ms.ooo.oooMmappedChunks)-1].ref
+			mmapRefs = []chunks.ChunkDiskMapperRef{ms.ooo.oooMmappedChunks[len(ms.ooo.oooMmappedChunks)-1].ref}
 		}
-		seq, off := mmapRef.Unpack()
+		lastMmapRef := mmapRefs[len(mmapRefs)-1]
```

Comment:
I don't know if this can fail, but it seems safer to just set the `lastMmapRef` to 0 if `len(mmapRefs)==0`.

Reply:
Do you mean after the if statement where if the length of the list of mmapRegs is 0, then the oooMmappedChunk list is checked?

Reply:
yes, in the original code the value of the reference was initialized to 0 (unknown), but here `mmapRefs[len(mmapRefs)-1]` can panic